mailarchive/mailarchive-subetha/src/test/org/sakaiproject/mailarchive/SplitEmailAddressTest.java

```diff
@@ -14,6 +14,30 @@ public void testGood() {
         assertEquals("example.com", email.getDomain());
     }
 
+    @Test
+    public void testBATVGood() {
+        // Check that our batv parsing works.
+        SplitEmailAddress email = SplitEmailAddress.parse("prvs=2987A7B7C7=me@example.com");
+        assertEquals("me", email.getLocal());
+        assertEquals("example.com", email.getDomain());
+    }
+
+    @Test
+    public void testBATVBad() {
+        // Check that our batv parsing is strict along the lines of the RFC
+        SplitEmailAddress email = SplitEmailAddress.parse("prvs=aaaaaaaaaa=me@example.com");
+        assertEquals("prvs=aaaaaaaaaa=me", email.getLocal());
+        assertEquals("example.com", email.getDomain());
+    }
+
+    @Test
+    public void testBATVSubAddress() {
+        // Check that we also catch the subaddressing style.
+        SplitEmailAddress email = SplitEmailAddress.parse("me+prvs=2987A7B7C7@example.com");
```

Comment:
me+prvs=aaaaaaaaaa@example.com is not stripped currently at the moment I'm just implementing a strict parser that uses the specification from the RFC. So it will only remove the BATV section if it matches the private signing format of:

    privs=(digit * 4)(hexnumber * 6)